src/Models/ExampleModel.php

```diff
@@ -0,0 +1,19 @@
+<?php
+
+declare(strict_types=1);
+
+
+namespace App\Models;
+
+
+use PDO;
+
+class ExampleModel
+{
+    protected PDO $db;
+
+    public function __construct(PDO $db)
```

Comment:
is this where the magical dependency injection happens?

Reply:
yes

Reply:
Yeah - Rayna are you familiar with Reflection classes?

The way it works is the autowirer makes a 'reflection' of the class you want, and it's able to use that reflection to figure out what the types of the constructor params are, so it knows what the deps are and is able to find them automatically